### PR TITLE
dbeaver/pro#2419 Fix multiple creation of parsing jobs

### PR DESCRIPTION
**Pull request by E1izabeth**
- [ ] No more than one parsing job
- [ ] Job name enhanced (contains script name and information about metadata search) - see video

https://github.com/dbeaver/dbeaver/assets/28875055/8d7b9da7-a5b7-43f2-a653-c7f909f81a5b

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java [hunk 1]
```diff
@@ -224,28 +236,16 @@ private void schedule(@Nullable DocumentEvent event) {
             }
 
             // TODO should we really schedule a new task each time this method called? or maybe at least cancel it at first
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java]: Is this TODO actual?

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java [hunk 2]
```diff
@@ -354,7 +353,14 @@ public Interval runWithResult() {
...
             return;
         }
-        IProgressMonitor monitor = Job.getJobManager().createProgressGroup();
+        
+        String textTitle = UIUtils.syncExec(new RunnableWithResult<String>() {
+            @Override
+            public String runWithResult() {
+                return editor.getTitle();
+            }
+        });
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java]: I don't quite understand the purpose of this code. It should be as simple as

```suggestion
        String textTitle = editor.getTitle();
```

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java (hunk 1 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java]: No

File: plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java (hunk 2 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java]: Job is executing not in the UI thread

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java]: But this method isn't required to be called on the UI thread

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/semantics/SQLBackgroundParsingJob.java]: Ok, I can remove it.
Are there any references of the workbench part methods being thread-safe?

**Comment by MashaKorax** [comment]
verified

**Pull request merged**